File: leetcode162.py
class Solution(object):
    def findPeakElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ls = len(nums)
        if ls == 1:
            return 0
        if nums[0] > nums[1]:
            return 0
        elif nums[-1]>nums[-2]:
            return ls-1
        for i in range(1,ls-1):
            if nums[i] > nums[i-1] and nums[i] > nums[i+1]:
                return i

File: leetcodec++/leetcode162.cpp
//
// Created by 黄敏辉 on 2019-08-21.
//

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        int low = 0, high = nums.size() - 1;
        while(low < high - 1){
            int mid = (low + high) / 2;
            if (nums[mid] > nums[mid - 1] && nums[mid] > nums[mid + 1])
                return mid;
            else if (nums[mid] > nums[mid + 1])
                high = mid - 1;
            else
                low = mid + 1;
        }

        return nums[low] > nums[high] ? low : high;
    }
};